### PR TITLE
Drop IE7 JS support

### PR DESCRIPTION
**Pull request by robinwhittleton**
IE8 supports JSON natively, so this is only in there for IE7 and lower. For services using govuk_frontend I don’t think there’s any expectation that JS will work in IE7.

This also fixes a warning in `gulp lint:scripts`.

### CODE REVIEW

**Comment by dsingleton** [comment]
👍 to dropping JS for IE7, and this lib.

Should we make that explicit and exempt IE7 from the layout templates? Eg, at https://github.com/alphagov/govuk_frontend_alpha/blob/master/src/templates/govuk_template.njk#L89

**Comment by robinwhittleton** [comment]
Yes, I’d be happy with that. I’ll convert this PR to a general ‘Drop IE7’ one.

**Comment by robinwhittleton** [comment]
So this patch now has two commits that effectively remove core JS from IE7 (we’re leaving the HTML5 shim in place so that it receives styles).

**Comment by dsingleton** [comment]
Added the start of some doc on this, otherwise i think this is 👍

File: docs/browser-support.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+   Over the last year we've seen a pretty dramatic drop in usage of legacy Internet Explorer, by at least half in most cases. Usage for IE 7, 8, 9 and 10 are all at ~.8% of traffic as of November, and IE 6 is less than 0.01%.
+
+   This makes a strong case for moving some, if not all, legacy versions of IE in to the _functional_ category of support.
+
+2. Reevaluating how we apply _functional_ for legacy browsers:
+
+  > If a browser is assigned a ‘functional’ level of support, it means your service might not look perfect in that browser but must still be usable.
+
+  Historically it's meant that a page looks same, but might not be pixel-perfect.
```
Comment (gemmaleigh) [inline docs/browser-support.md]: I think there's a 'the' missing between looks and same.

File: docs/browser-support.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  As usage stats change, our approach to high-profile but low usage browsers can be more relaxed, eg, potentially serving grid content in a linear/mobile view.
+
+
+### Dropping JS behaviour for IE6 and IE7
+
+We've chosen to not provide any _behavioural_ JS to the oldest versions if Internet Explorer - though we're still providing the [HTML5 Shim](https://github.com/aFarkas/html5shiv), to ensure HTML 5 tags are style-able.
+
+As GOV.UK practices progressive enhancement all content should be usable regardless of JS, which fits the definition of _functional_.
```
Comment (gemmaleigh) [inline docs/browser-support.md]: Worth linking to the progressive enhancement blog post here?

**Comment by dsingleton** [comment]
Fixed the doc comments - happy with everything else. OK to merge?

**Pull request merged**